src/lib/autodiscovery/autodiscovery.py

```diff
@@ -0,0 +1,110 @@
+import os
+import time
+from dataclasses import asdict
+from typing import List
+
+from aiohttp import ClientSession
+
+from lib.autodiscovery.gcp_metrics_descriptor import GCPMetricDescriptor
+from lib.clientsession_provider import init_gcp_client_session
+from lib.configuration import config
+from lib.context import LoggingContext
+from lib.credentials import create_token
+from lib.metrics import GCPService, Metric
+
+logging_context = LoggingContext("AUTODISCOVERY")
+
+
+discovered_resource_type = os.environ.get("AUTODISCOVERY_RESOURCE_TYPE", "gce_instance")
```

Comment:
taking this from env var is like dev only tool right? Probably should not be merged

Reply:
This is only because this ticket is for only one extension, and in this env we can change it. So It should be removed? Hardcoded? or something else. Because like i said it is still on /feature/autodiscovery branch not the main one.

Reply:
Are we not doing this now for all extensions we have? Or enabled ones?

Reply:
> Are we not doing this now for all extensions we have? Or enabled ones?

For any chosen 1 extension that is enabled

Reply:
Yes, ok. I just meant that I understood that for this first iteration we wanted to go with, let's say, all enabled extensions.